dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js

```diff
@@ -0,0 +1,302 @@
+
+    var longestDuration = _.max(dataSets
+        .filter(function (data) {
+          return !_.includes(disabledDataSets, data);
+        })
+        .map(function (data) {
+          return data.donors.slice(-1)[0].survivalTime;
+        }));
+
+    x.domain([0, longestDuration]);
+    y.domain([0, 1]);
+
+    // draw x axis
+    svg.append('g')
+      .attr('class', 'x axis')
+      .attr('transform', 'translate(' + margin.left + ',' + (axisHeight + margin.top) + ')')
+      .call(xAxis)
+      .append('text')
+        .attr('class', 'axis-label')
+        .attr('dy', 30)
+        .attr('x', axisWidth / 2)
+        .style('text-anchor', 'end')
+        .text(xAxisLabel);
+
+    // draw y axis
+    svg.append('g')
+      .attr('class', 'y axis')
+      .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')')
+      .call(yAxis)
+      .append('text')
+        .attr('class', 'axis-label')
+        .attr('transform', 'rotate(-90)')
+        .attr('y', -30)
+        .attr('x', - (margin.top + axisHeight / 2))
+        .text(yAxisLabel);
+
+    dataSets.forEach(function (data, i) {
+      if (_.includes(disabledDataSets, data)) {
+        return;
+      }
+      var line = d3.svg.area()
+        .interpolate('step-after')
+        .x(function(p) { return x(p.x); })
+        .y(function(p) { return y(p.y); });
+
+      var setGroup = svg.append('g')
+        .attr('class', 'serie')
+        .attr('set-id', data.meta.id)
+        .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')');
+      var setColor = palette[i % palette.length];
+
+      // draw the data as an svg path
+      setGroup.append('path')
+        .datum(data.donors.map(function (d) { return {x: d.survivalTime, y: d.survivalEstimate}; }))
+        .attr('class', 'line')
+        .attr('d', line)
+        .attr('stroke', setColor);
+
+      // draw the data points as circles
+      setGroup.selectAll('circle')
+        .data(data.donors)
+        .enter()
+        .append('svg:circle')
+          .attr('class', 'point')
+          .attr('status', function (d) { return d.status; })
+          .attr('cx', function(d) { return x(d.survivalTime); })
+          .attr('cy', function(d) { return y(d.survivalEstimate); })
+          .attr('fill', setColor )
+          .on('mouseover', function (d) {
+            onMouseEnterDonor(d3.event, d);
+          })
+          .on('mouseout', function (d) {
+            onMouseLeaveDonor(d3.event, d);
+          })
+          .on('click', function (d) {
+            onClickDonor(d3.event, d);
+          })
+        ;
+    });
+
+    return svg;
+  }
+
+  var survivalAnalysisController = [
+    '$scope',
+    '$element',
+    function (
+      $scope,
+      $element
+    ) {
+      var ctrl = this;
+      var graphContainer = $element.find('.survival-graph').get(0);
+      var svg = d3.select(graphContainer).append('svg');
+      var tipTemplate = _.template($element.find('.survival-tip-template').html());
```

Comment:
Technically all of this should probably go in a directive since it dom manipulation heavy. If you did that, you would have a natural place to hook into the directive template. It seems as though this has been a common pattern in our code base at least.